src/context/TodosContext.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { createContext, useContext } from 'react';
+import { UseTodos, useTodos } from '../hooks/useTodos';
+import { temporaryTodo } from '../utils/tempTodo';
+
+interface TodosProviderProps {
+  children: React.ReactNode;
+}
+
+const initialContextValue: UseTodos = {
+  todos: [],
+  setTodos: () => {},
+  errorMessage: '',
+  setErrorMessage: () => {},
+  tempTodo: temporaryTodo,
+  setTempTodo: () => {},
+  loadingIds: [],
+  setLoadingIds: () => {},
+};
+
+const TodosContext = createContext<UseTodos>(initialContextValue);
+
+export const TodosProvider: React.FunctionComponent<TodosProviderProps> = ({
+  children,
+}) => {
+  const todosData = useTodos();
```

Comment:
I don't see the point in `useTodos` hook
It just adds unnecessary complications without giving you anything.

As long as you use it __only__ inside this context, you can simply throw away this hook and move all code directly to the context.

Reply:
If you decide keeping it, it makes sense at least move it to the context/TodosContext folder, as it relates only to the context